### DISCUSSION TITLE
Replacement preset for `actions/upload-artifact`and `actions/upload-artifact` v1 and v2 to v4

### DISCUSSION THREAD

**Discussion by HonkingGoose** [renovatebot/renovate#27289] [Suggest an Idea]
### Tell us more.

The v1 and v2 versions of `actions/upload-artifact` and `actions/upload-artifact` will be deprecated on June 30, 2024: [^1]

> With the introduction of v4, we will be deprecating v1 and v2 of actions/upload-artifact, actions/download-artifact, and related npm packages on June 30, 2024. We strongly encourage customers to update their workflows to begin using v4 of the artifact actions.

Do we want a replacement preset preset [^2] for the `actions/upload-artifact`and `actions/upload-artifact` v1 and v2 to v4?

The GitHub blog says v4 is way faster. [^1] So if we're going to make a preset, I think we should "skip" v3 and propose the v4 upgrade:

> On December 14, 2023, GitHub Actions released [v4 of the actions to upload and download artifacts](https://github.blog/2024-02-12-get-started-with-v4-of-github-actions-artifacts/). This version improves upload/download speeds by up to 98%, addresses long-standing customer feedback requests, and represents the future of artifacts in GitHub Actions.

Let me know what you guys think. 😄 

[^1]: [GitHub Blog, Deprecation notice: v1 and v2 of the artifact actions](https://github.blog/changelog/2024-02-13-deprecation-notice-v1-and-v2-of-the-artifact-actions/)
[^2]: [Renovate docs, replacement presets](https://docs.renovatebot.com/presets-replacements/)

**Comment by rarkins** [comment]
It's just an upgrade though isn't it? Not a replacement?

**Reply by viceice** [reply]
yes, and renovate will alsways suggest the latest major by default. Users will be warned in workflows before they break. so i don't think we need to do anything here.

**Reply by HonkingGoose** [reply]
GitHub will break the old actions after the deprecation:

> In order to prevent issues for customers using GitHub Connect, the tags for v1 through v2 will not be removed from the actions/upload-artifact and actions/download-artifact project repositories. However, attempting to use a version of the actions after the announced deprecation date will result in a workflow failure. This deprecation will not impact any existing versions of GitHub Enterprise Server being used by customers.

**Reply by viceice** [reply]
yes, but renovate will already suggest the major update by default.

**Reply by HonkingGoose** [reply]
We'll leave things as they are then. 😄 Thank you for the quick responses! ❤️

**Reply by viceice** [reply]
btw: GHES users would probably broken when we force the v4 update.

**Reply by HonkingGoose** [reply]
> This deprecation will not impact any existing versions of GitHub Enterprise Server being used by customers.

@viceice The way I'm reading this is that GHES will keep working, but "standard GitHub" will break.

**Reply by viceice** [reply]
yes, i know. but if we add a replacement, than it will open a replacement pr for all. even on ghes and gitea / forgejo.
